pkg/manager/manager.go

```diff
@@ -580,6 +580,14 @@ func defaultBaseContext() context.Context {
 	return context.Background()
 }
 
+// NewProvider use spec managerBroadcaster to replace cluster's makeBroadcaster.
+func NewProvider(config *rest.Config, httpClient *http.Client, scheme *runtime.Scheme, logger logr.Logger, makeBroadcaster intrec.EventBroadcasterProducer) (*intrec.Provider, error) {
+	managerBroadcaster := func() (record.EventBroadcaster, bool) {
+		return record.NewBroadcaster(), true
+	}
+	return intrec.NewProvider(config, httpClient, scheme, logger, managerBroadcaster)
+}
```

Comment:
```
pkg/manager/manager.go
setOptionsDefaults()
if options.EventBroadcaster == nil {
		// defer initialization to avoid leaking by default
		options.makeBroadcaster = func() (record.EventBroadcaster, bool) {
			return record.NewBroadcaster(), true
		}
	} else {
		options.makeBroadcaster = func() (record.EventBroadcaster, bool) {
			return options.EventBroadcaster, false
		}
	}
```
i see options can use EventBroadcaster.

Reply:
Could we actually add a test first that shows the issue you're referring to? I'm still not sure how does the change above actually solve the issue

Reply:
@niuyueyang1996 with you're code we still use the EventBroadCaster to build this function and set it in makeBroadcaster, but makeBroadcaster is not used anymore